pkg/sync/destination.go

```diff
@@ -54,6 +59,19 @@ func NewImageDestination(registry, repository, tag, username, password string, i
 
 	ctx := context.WithValue(context.Background(), ctxKey{"ImageDestination"}, repository)
 	if username != "" && password != "" {
+		fmt.Printf("Credential processing for %s/%s ...\n", registry, repository)
+		if strings.Contains(registry, ".gcr.io") {
```

Comment:
It seems that this commit changes all the behaviors of synchronization about `"*.gcr.io"` registry, which seems dangerous.

So I'm wondering if it's a better way that, we try to get oauth2 token only if the `username` is `"oauth2accesstoken"` and the `registry` string contains `".gcr.io"`. Then the old behaviors will not be affected.

Reply:
Have added code for that. The "username" is checked against "_json_key" which identifies the permament service account token.